source/extensions/dynamic_modules/abi.h

```diff
@@ -262,13 +262,15 @@ typedef enum {
  */
 typedef enum {
   // stream's dynamic metadata.
-  envoy_dynamic_module_type_metadata_source_dynamic,
+  envoy_dynamic_module_type_metadata_source_Dynamic,
   // route metadata
-  envoy_dynamic_module_type_metadata_source_route,
+  envoy_dynamic_module_type_metadata_source_Route,
   // cluster metadata
-  envoy_dynamic_module_type_metadata_source_cluster,
+  envoy_dynamic_module_type_metadata_source_Cluster,
```

Comment:
note: i fixed the style as I explained in the PR description to match other places, e.g.

* https://github.com/envoyproxy/envoy/blob/8f31220e39797ae05e2056c076f3d99d9ee48170/source/extensions/dynamic_modules/abi.h#L189-L257
* https://github.com/envoyproxy/envoy/blob/8f31220e39797ae05e2056c076f3d99d9ee48170/source/extensions/dynamic_modules/abi.h#L407-L442

Reply:
This style is a little weird. Is there any explanation about why this stype is used?

Reply:
This matches the one we used for C++ (and Rust at SDK level) - variants are Pascal case. Unfortunately in C, the variants won't be namespaced, so we need this name prefix before the Pascal case...